src/Spago/PackageSet.hs

```diff
@@ -40,6 +48,112 @@ makePackageSetFile force comments = do
     else logWarn $ display $ Messages.foundExistingProject packagesPath
   Dhall.format packagesPath
 
+-- | Use the specified version of the package set (if specified).
+--   Otherwise, get the latest version of the package set if possible
+updatePackageSetVersion
+  :: forall env
+  .  (HasLogFunc env, HasGlobalCache env)
+  => Maybe Text
+  -> RIO env ()
+updatePackageSetVersion = maybe useLatestRelease useSpecificRelease
+
+useSpecificRelease
```

Comment:
I feel like we could reduce code duplication a bit here, as it's exactly the same in many places so it might be really confusing to maintain - I would like to:
- add a `where` block in `updatePackageSetVersion`
- inline in that block `useSpecificRelease`, `useLatestRelease`, `updateTag`, `getCurrentTag`, `upgradeImports`.
  In this way we'd only get one definition for these subroutines. It's also a good occasion to remove the old code that has the `TODO: remove` there (i.e. we should keep the new definitions you added, not the old ones)

Reply:
Done.